### DISCUSSION TITLE
How to use default constructor for nested model

### DISCUSSION THREAD

**Discussion by davetapley** [mobxjs/mobx-state-tree#1908] [Q&A] [answered]
If I have an `array`, I can call `create()` and get empty array:

```ts
const Thing = types.model({idx: types.identifier});
const Store = types.model({things: types.array(Thing)})

const store = Store.create();
```

But if I have a parent object:

```ts
const Thing = types.model({idx: types.identifier});
const Store = types.model({things: types.array(Thing)})
const Root = types.model({thingStore: Store})

const root = Root.create();

// Error while converting `{}` to `AnonymousModel`
```

I can fix with:
```ts
Root.create({ thingStore: {} });
```

But it would be nice if the the default empty constructor would be invoked automatically.

Am I missing something? 🤔

**Comment by EmilTholin** [comment, marked as answer]
Hi @davetapley!

As the [`Miscellaneous Tips` part of the documenatation states](https://mobx-state-tree.js.org/tips/more-tips#optional-empty-maps-arrays) `types.array(Thing)` is the same thing as `types.optional(types.array(Thing), [])`, but that's not the case for `types.model`.

You have to explicitly set it to `types.optional(Store, {})` for it to work the way you would like:

```js
const Thing = types.model({ idx: types.identifier });
const Store = types.model({ things: types.array(Thing) });
const Root = types.model({ thingStore: types.optional(Store, {}) });

const root = Root.create();

// Success!
```

**Reply by davetapley** [reply]
Thank you!

I think I was misled by `optional` because I was thinking of it as optional in the _maybe undefined_ sense, not in the _may be omitted from the snapshot_ sense.
